tmail-backend/mailbox/encrypted/memory/src/main/java/com/linagora/tmail/encrypted/InMemoryEncryptedEmailContentStore.scala

```diff
@@ -60,16 +60,15 @@ class InMemoryEncryptedEmailContentStore @Inject()(blobStore: BlobStore,
     val position: AtomicInteger = new AtomicInteger(POSITION_NUMBER_START_AT)
     SFlux.fromIterable(encryptedAttachmentContents)
       .concatMap(attachmentContent => SMono.fromPublisher(blobStore.save(bucketName, attachmentContent.getBytes(StandardCharsets.UTF_8), storagePolicy)))
-      .map(blobId=> position.getAndIncrement()-> blobId)
-      .collectSeq()
-      .map(positionBlobIdList=> positionBlobIdList.toMap)
+      .map(blobId => position.getAndIncrement() -> blobId)
+      .collectMap(positionBlobId => positionBlobId._1, positionBlobId => positionBlobId._2)
       .map(positionBlobIdMap => messageIdBlobIdStore.put(messageId, positionBlobIdMap))
       .`then`()
   }
 
   private def removeAttachment(messageId: MessageId): SMono[Unit] =
     deleteBlobStore(messageId)
-      .collectSeq()
+      .`then`()
```

Comment:
not working with just `then(SMono.just(messageIdBlobIdStore.remove(messageId)))`? Do you need two `then`?

Reply:
Yes, we need, I tried with single `then`, but I can not, `deleteBlobStore(messageId)` return Flux, so We need one `then` in middle to convert